From: =?utf-8?q?Christian_G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Sun, 5 Nov 2017 14:20:17 +0100
Subject: vnstat.service: add 3s timeout for auto-restart

---
 examples/systemd/vnstat.service | 1 +
 1 file changed, 1 insertion(+)

diff --git a/examples/systemd/vnstat.service b/examples/systemd/vnstat.service
index d51c45c..a32e342 100644
--- a/examples/systemd/vnstat.service
+++ b/examples/systemd/vnstat.service
@@ -8,6 +8,7 @@ ExecStart=/usr/sbin/vnstatd -n
 ExecReload=/bin/kill -HUP $MAINPID
 Restart=on-failure
 User=vnstat
+RestartSec=3
 
 # Hardening
 ProtectSystem=strict
