From: =?utf-8?q?Christian_G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Thu, 24 Aug 2017 13:44:26 +0200
Subject: fix Wimplicit-function-declaration on kFreeBSD

---
 src/misc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/misc.c b/src/misc.c
index 311977b..6b7542d 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -4,6 +4,7 @@
 /* enable wcswidth on kFreeBSD */
 #if defined(__FreeBSD_kernel__) && defined(__GLIBC__)
 #define __USE_XOPEN
+#define _XOPEN_SOURCE
 #endif
 #include "common.h"
 #include "misc.h"
