From: =?utf-8?q?Christian_G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Thu, 24 Aug 2017 15:52:51 +0200
Subject: fix gcc Wdiscarded-qualifiers warnings in testsuite

---
 tests/misc_tests.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/misc_tests.c b/tests/misc_tests.c
index 7815bdf..3bad42a 100644
--- a/tests/misc_tests.c
+++ b/tests/misc_tests.c
@@ -11,7 +11,7 @@ END_TEST
 
 START_TEST(getunitprefix_returns_something_with_all_cfg_combinations)
 {
-	char *string;
+	const char *string;
 	int j;
 
 	cfg.unitmode = _i;
@@ -24,7 +24,7 @@ END_TEST
 
 START_TEST(getrateunitprefix_returns_something_with_all_cfg_combinations)
 {
-	char *string;
+	const char *string;
 	int j;
 
 	for (j=1; j<=(UNITPREFIXCOUNT+1); j++) {
