From: =?utf-8?q?Christian_G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Tue, 5 Sep 2017 11:23:17 +0200
Subject: fix Time of check time of use issues

Found by coverity
---
 src/fs.c | 21 ++++++++++++++++-----
 1 file changed, 16 insertions(+), 5 deletions(-)

diff --git a/src/fs.c b/src/fs.c
index abf0035..11d7634 100644
--- a/src/fs.c
+++ b/src/fs.c
@@ -166,6 +166,7 @@ void updatedirownerid(const char *dir, const uid_t uid, const gid_t gid)
 	struct dirent *di;
 	struct stat statbuf;
 	char entryname[512];
+	int dir_fd, file_fd;
 
 	if (!cfg.updatefileowner) {
 		return;
@@ -177,14 +178,18 @@ void updatedirownerid(const char *dir, const uid_t uid, const gid_t gid)
 		return;
 	}
 
-	if (stat(dir, &statbuf)!=0) {
+	if ((dir_fd = open(dir, O_RDONLY | O_CLOEXEC)) == -1)
+		return;
+	if (fstat(dir_fd, &statbuf)!=0) {
+		close(dir_fd);
 		return;
 	}
 
 	if (statbuf.st_uid != uid || statbuf.st_gid != gid) {
-		if (chown(dir, uid, gid) != 0) {
+		if (fchown(dir_fd, uid, gid) != 0) {
 			if (debug)
 				printf("Error: updatedirowner() chown() \"%s\": %s\n", dir, strerror(errno));
+			close(dir_fd);
 			return;
 		} else {
 			if (debug)
@@ -192,9 +197,10 @@ void updatedirownerid(const char *dir, const uid_t uid, const gid_t gid)
 		}
 	}
 
-	if ((d=opendir(dir))==NULL) {
+	if ((d=fdopendir(dir_fd))==NULL) {
 		if (debug)
 			printf("Error: updatedirowner() diropen() \"%s\": %s\n", dir, strerror(errno));
+		close(dir_fd);
 		return;
 	}
 
@@ -203,11 +209,14 @@ void updatedirownerid(const char *dir, const uid_t uid, const gid_t gid)
 			continue;
 		}
 		snprintf(entryname, 512, "%s/%s", dir, di->d_name);
-		if (stat(entryname, &statbuf)!=0) {
+		if ((file_fd = open(entryname, O_RDONLY | O_CLOEXEC)) == -1)
+			continue;
+		if (fstat(file_fd, &statbuf)!=0) {
+			close(file_fd);
 			continue;
 		}
 		if (statbuf.st_uid != uid || statbuf.st_gid != gid) {
-			if (chown(entryname, uid, gid) != 0) {
+			if (fchown(file_fd, uid, gid) != 0) {
 				if (debug)
 					printf("Error: chown() \"%s\": %s\n", entryname, strerror(errno));
 			} else {
@@ -215,9 +224,11 @@ void updatedirownerid(const char *dir, const uid_t uid, const gid_t gid)
 					printf("\"%s\" chown completed\n", entryname);
 			}
 		}
+		close(file_fd);
 	}
 
 	closedir(d);
+	close(dir_fd);
 }
 
 void matchdbownerwithdirowner(const char *dir)
