--- a/Makefile
+++ b/Makefile
@@ -35,30 +35,12 @@ debug:
 install:
 	@echo "Installing vnStat..."
 
-# check that system is really Linux
-	@if [ `uname` != "Linux" ]; \
-	then echo "This isn't a Linux system. Maybe 'make bsdinstall' is what you need?"; \
-	false; \
-	fi
-
 # check that there's something to install
 	@if [ ! -f "src/vnstat" ] || [ ! -f "src/vnstatd" ]; \
 	then echo "Nothing to install, run make first."; \
 	false; \
 	fi
 
-# move some really old version database(s) if found
-	@if [ -d "$(DESTDIR)/var/spool/vnstat" ]; \
-	then echo "Moving old database(s) to new location..."; \
-	mv -f $(DESTDIR)/var/spool/vnstat $(DESTDIR)/var/lib/; \
-	fi
-
-# remove some really old version binary if found
-	@if [ -x "$(DESTDIR)/usr/local/bin/vnstat" ]; \
-	then echo "Removing old binary..."; \
-	rm -f $(DESTDIR)/usr/local/bin/vnstat; \
-	fi
-
 # install default config if such doesn't exist
 	@if [ ! -f "$(ETC)/vnstat.conf" ]; \
 	then echo "Installing config to $(ETC)/vnstat.conf"; \
@@ -84,23 +66,6 @@ install:
 	install -m 644 man/vnstati.1 $(MAN)/man1; \
 	fi
 
-	@if [ -f "$(MAN)/man1/vnstat.1.gz" ]; \
-	then gzip -f9 $(MAN)/man1/vnstat.1; \
-	gzip -f9 $(MAN)/man1/vnstatd.1; \
-	gzip -f9 $(MAN)/man5/vnstat.conf.5; \
-	if [ -f "src/vnstati" ]; \
-	then gzip -f9 $(MAN)/man1/vnstati.1; \
-	fi; \
-	fi
-
-# remove vnstat.conf.1 is such exists in the wrong place
-	@if [ -f "$(MAN)/man1/vnstat.conf.1.gz" ]; \
-	then rm -f $(MAN)/man1/vnstat.conf.1.gz; \
-	fi
-	@if [ -f "$(MAN)/man1/vnstat.conf.1" ]; \
-	then rm -f $(MAN)/man1/vnstat.conf.1; \
-	fi
-
 	@echo " "
 	@echo "No service file or startup script has been installed. See the"
 	@echo "INSTALL document for instructions on how to enable vnStat."
