--- a/tests/Makefile
+++ b/tests/Makefile
@@ -3,7 +3,7 @@ CC = gcc
 endif
 CFLAGS ?= -Wall -Wextra -g
 CFLAGS += -DPROCNETDEV=\"testdir/proc/dev\" -DSYSCLASSNET=\"testdir/sysclassnet\"
-LDLIBS = -lm -lcheck -lrt -lpthread
+LDLIBS = -lm -lcheck -lrt -lpthread -lsubunit
 OBJS = vnstat_tests.o common_tests.o database_tests.o config_tests.o ifinfo_tests.o misc_tests.o daemon_tests.o
 OBJS += common.o ifinfo.o dbshow.o dbaccess.o dbcache.o cfg.o ibw.o misc.o daemon.o
 
