From: Felix Geyer <fgeyer@debian.org>
Date: Sun, 20 Aug 2017 11:18:29 +0200
Subject: Adjust PidFile path to the new location

---
 cfg/vnstat.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cfg/vnstat.conf b/cfg/vnstat.conf
index c127b29..a99420a 100644
--- a/cfg/vnstat.conf
+++ b/cfg/vnstat.conf
@@ -118,7 +118,7 @@ UpdateFileOwner 1
 LogFile "/var/log/vnstat/vnstat.log"
 
 # file used as daemon pid / lock file
-PidFile "/var/run/vnstat/vnstat.pid"
+PidFile "/run/vnstat/vnstat.pid"
 
 
 # vnstati
