From: Felix Geyer <fgeyer@debian.org>
Date: Sun, 20 Aug 2017 11:18:29 +0200
Subject: Run vnstatd as user vnstat

---
 examples/systemd/vnstat.service | 1 +
 1 file changed, 1 insertion(+)

diff --git a/examples/systemd/vnstat.service b/examples/systemd/vnstat.service
index 9a39994..d51c45c 100644
--- a/examples/systemd/vnstat.service
+++ b/examples/systemd/vnstat.service
@@ -7,6 +7,7 @@ After=network.target
 ExecStart=/usr/sbin/vnstatd -n
 ExecReload=/bin/kill -HUP $MAINPID
 Restart=on-failure
+User=vnstat
 
 # Hardening
 ProtectSystem=strict
